src/sentry/utils/metrics.py

```diff
@@ -58,6 +60,18 @@ def _add_global_tags(_all_threads: bool = False, **tags: TagValue) -> List[Tags]
     return stack
 
 
+def _filter_tags(tags: MutableTags) -> MutableTags:
+    """Removes unwanted tags from the tag mapping and returns a filtered one."""
+    discarded = frozenset(key for key in tags if key.endswith("_id") or key in BAD_TAGS)
```

Comment:
One other thing we could add, since ints are rarely valid:
```
ALLOWED_NUMERIC_KEYS = frozenset(['status_code'])

def is_invalid_number_value(key, value):
    is_number = isinstance(value, (int, float))
    if isinstance(value, str):
        is_number = value.isdigit()
    return is_number or key not in ALLOWED_NUMERIC_KEYS
     

frozenset(key for (key, value) in tags.items() if key.endswith("_id") or key in BAD_TAGS or is_invalid_number_value(key, value))
```

Maybe just as a follow up or something

Reply:
I was considering adding that but that might require more auditing of what we are currently emitting.